components/host-dashboard/reports-section/TransactionsOverviewSection.js

```diff
@@ -1,13 +1,117 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { FormattedMessage } from 'react-intl';
+import dynamic from 'next/dynamic';
+import { FormattedMessage, useIntl } from 'react-intl';
 
 import { formatCurrency } from '../../../lib/currency-utils';
+import { days } from '../../../lib/utils';
 
-import { Box } from '../../Grid';
+import Container from '../../Container';
+import { Box, Flex } from '../../Grid';
 import LoadingPlaceholder from '../../LoadingPlaceholder';
 import ProportionalAreaChart from '../../ProportionalAreaChart';
 import { P, Span } from '../../Text';
+const Chart = dynamic(() => import('react-apexcharts'), { ssr: false });
+
+const getChartOptions = (intl, startDate, endDate, hostCreatedAt) => {
+  return {
+    chart: {
+      id: 'chart-transactions-overview',
+    },
+    legend: {
+      show: true,
+      horizontalAlign: 'left',
+    },
+    colors: ['#29CC75', '#F55882'],
+    grid: {
+      xaxis: { lines: { show: true } },
+      yaxis: { lines: { show: false } },
+    },
+    stroke: {
+      curve: 'straight',
+      width: 1.5,
+    },
+    dataLabels: {
+      enabled: false,
+    },
+
+    xaxis: {
+      categories: getCategories(intl, startDate, endDate, hostCreatedAt),
+    },
+    yaxis: {
+      labels: {
+        formatter: function (value) {
+          return value < 1000 ? value : `${Math.round(value / 1000)}k`;
+        },
+      },
+    },
+  };
+};
+
+const getCategories = (intl, startDate, endDate, hostCreatedAt) => {
+  const numberOfDays = days(startDate || hostCreatedAt, endDate);
+  if (numberOfDays <= 7) {
+    const startDay = startDate.getDay();
+    return [...new Array(7)].map(
+      (_, idx) => `${intl.formatDate(new Date(0, 0, idx + startDay), { weekday: 'long' }).toUpperCase()}`,
+    );
+  } else if (numberOfDays <= 365) {
+    const currentMonth = new Date().getMonth();
+    return [...new Array(12)].map(
+      (_, idx) => `${intl.formatDate(new Date(0, idx + currentMonth + 1), { month: 'short' }).toUpperCase()}`,
+    );
+  } else {
+    return [...new Array(6)].map(
+      (_, idx) =>
+        `${intl.formatDate(new Date(new Date().getFullYear() - 5 + idx, 0), { year: 'numeric' }).toUpperCase()}`,
+    );
+  }
+};
+
+const getCategoryType = (startDate, endDate, hostCreatedAt) => {
+  const numberOfDays = days(startDate || hostCreatedAt, endDate);
+  if (numberOfDays <= 7) {
+    return 'WEEK';
+  } else if (numberOfDays <= 365) {
+    return 'MONTH';
+  } else {
+    return 'YEAR';
+  }
+};
+
+const constructDataPointObjects = (category, dataPoints) => {
+  let dataPointObject;
+  if (category === 'YEAR') {
+    dataPointObject = new Array(6).fill(0);
+    const currentYear = new Date().getFullYear();
+    dataPoints.forEach(dataPoint => {
+      const year = new Date(dataPoint.date).getFullYear();
+      if (year > currentYear - 6) {
+        dataPointObject[5 - (currentYear - year)] = dataPoint.amount.value;
+      }
```

Comment:
This is difficult to parse. Can we add some comments? Also for the other ones (month, week)

Reply:
I've added some comments to explain what we are doing. Let me know if that's enough or if we need more clarification. 🤔